packages/contracts/scripts/deploy/dollar/solidityScripting/02_UbiquityDollarToken.s.sol

```diff
@@ -1,30 +1,40 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.16;
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./01_Diamond.s.sol";
 
-import "./01_UbiquityDollarManager.s.sol";
-
-contract DollarScript is ManagerScript {
+contract DollarScript is DiamondScript {
     UbiquityDollarToken dollar;
     address metapool;
 
     function run() public virtual override {
         super.run();
         vm.startBroadcast(deployerPrivateKey);
 
-        dollar = new UbiquityDollarToken(manager);
-        manager.setDollarTokenAddress(address(dollar));
-
-        dollar.mint(address(manager), 10000e18);
-
-        manager.deployStableSwapPool(
+        dollar = new UbiquityDollarToken(address(diamond));
+        IManager.setDollarTokenAddress(address(dollar));
+        IAccessCtrl.grantRole(DOLLAR_TOKEN_MINTER_ROLE, address(diamond));
+        IAccessCtrl.grantRole(DOLLAR_TOKEN_BURNER_ROLE, address(diamond));
+
+        dollar.mint(address(diamond), 10000e18);
+        uint256 adminBal = IERC20(curve3PoolToken).balanceOf(admin);
+        console.log("----ADMIN 3CRV bal:", adminBal);
+        // deployer needs 10000  3CRV to deploy the pool
+        IERC20(curve3PoolToken).transfer(address(diamond), 10000e18);
+        uint256 diamondBal = IERC20(curve3PoolToken).balanceOf(
+            address(diamond)
+        );
+        console.log("----DIAMOND 3CRV bal:", diamondBal);
+        //  deal(curve3PoolToken, address(diamond), 10000e18);
+        IManager.deployStableSwapPool(
             curveFactory,
             basePool,
             curve3PoolToken,
             10,
             5000000
         );
 
-        metapool = manager.stableSwapMetaPoolAddress();
+        metapool = IManager.stableSwapMetaPoolAddress();
```

Comment:
Why do you prefix with `I` now?

Reply:
Like in the test IxxxFacet is the facet implementation through the diamond proxy whereas xxxFacet is the implementation of the facet (so not through the proxy)